### PR TITLE
bpf: Replace old school header guards with #pragma once

### PR DESCRIPTION
**Pull request by dylandreimerink**
This commit replaces the old school header guards with #pragma once. This is a more modern way of preventing multiple inclusion of the same header file.

In future we will be using scripts to remove macros, by replacing these now with the proper #pragma once we avoid having to write exceptions for these in the scripts.

### CODE REVIEW

**Review by gandro — APPROVED** [review]
Thanks!

**Comment by lmb** [comment]
Did you figure out a lint to avoid adding more of the old style guards? Do we need to update a style guide somewhere?

**Review by aspsk — APPROVED** [review]

**Review by youngnick — APPROVED** [review]

**Comment by dylandreimerink** [comment]
> Did you figure out a lint to avoid adding more of the old style guards?

Ah, forgot about that. Will add something, should be fairly doable. 

> Do we need to update a style guide somewhere?

I don't think we have one for the C code, I guess checkpatch does most of that.

File: tools/legacyhguardcheck/main.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		os.Exit(1)
+	}
+
+	found, err := checkDir(os.Args[1])
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	if found {
+		fmt.Fprintln(os.Stderr, "Found legacy header guards, please replace with #pragma once or "+
```
Comment (lmb) [inline tools/legacyhguardcheck/main.go:29]: What does the output look like? Does it include line and file number?

Comment (dylandreimerink) [reply tools/legacyhguardcheck/main.go:29]: Good point. I added a line number, it now looks like: `Found legacy header guard with __EXAMPLE_H__ macro at bpf/some_example_file.h:123`

Comment (dylandreimerink) [reply tools/legacyhguardcheck/main.go:29]: Also, to clarify. There are 2 messages. This global message with an instruction and the per-instance message at line 96

**Review by lmb — APPROVED** [review]

File: bpf/include/bpf/api.h [hunk 2]
```diff
@@ -19,4 +18,3 @@
 #include "csum.h"
 #include "access.h"
```
Comment (borkmann) [inline bpf/include/bpf/api.h]: small nit: please also remove trailing newline in this and other files

File: bpf/tests/tc_nodeport_lb_terminating_backend.c [hunk 3]
```diff
@@ -67,7 +67,6 @@ mock_ctx_redirect(const struct __sk_buff *ctx __maybe_unused,
 
 #define SECCTX_FROM_IPCACHE 1
 
-#include "config_replacement.h"
```
Comment (borkmann) [inline bpf/tests/tc_nodeport_lb_terminating_backend.c]: nit: two empty lines

**Review by borkmann — APPROVED** [review]
(ack, assuming small nits get fixed)

**Review by pchaigno — APPROVED** [review]

**Pull request merged**